torch/csrc/jit/codegen/cuda/lower_alias_memory.cpp

```diff
@@ -666,15 +670,19 @@ class BufferUseDefInfo {
     for (auto output_tv : ir_utils::filterByType<TensorView>(expr->outputs())) {
       auto maybe_alloc_info = getMaybeAllocInfoFromTV(output_tv);
       if (maybe_alloc_info.has_value()) {
+        // Reductions use outputs as read-write parameters, so their
+        // outputs need to be marked as read as well
+        const bool is_read_write = ir_utils::isReductionOp(expr);
```

Comment:
I think this should be fine when we remove trivial reduction, but for now should we specialize to not include trivial reduce?

Reply:
The alias analysis ignores size-1 (scalar) tensors, so it shouldn't matter.

Reply:
Well, sorry, trivial reductions don't always mean size-1. I'll update the code.